zoneinfo/src/posix.rs

```diff
@@ -0,0 +1,228 @@
+            offset,
+            transition_info: None,
+        }
+    }
+
+    pub(crate) fn from_zone_and_rules(entry: &ZoneEntry, rules: &LastRules) -> Self {
+        let offset = entry.std_offset.add(rules.standard.save);
+        let formatted = entry.format.format(
+            entry.std_offset.as_secs(),
+            rules.standard.letter.as_deref(),
+            rules.standard.is_dst(),
+        );
+        let is_numeric = is_numeric(&formatted);
+        let abbr = PosixAbbreviation {
+            is_numeric,
+            formatted,
+        };
+
+        let transition_info = rules.saving.as_ref().map(|rule| {
+            let formatted = entry.format.format(
+                entry.std_offset.as_secs() + rule.save.as_secs(),
+                rule.letter.as_deref(),
+                rule.is_dst(),
+            );
+            let abbr = PosixAbbreviation {
+                is_numeric,
+                formatted,
+            };
+            let savings = rule.save;
+            let start = PosixDateTime::from_rule_and_transition_info(
+                rule,
+                entry.std_offset,
+                rules.standard.save,
+            );
+            let end = PosixDateTime::from_rule_and_transition_info(
+                &rules.standard,
+                entry.std_offset,
+                rule.save,
+            );
+            PosixTransition {
+                abbr,
+                savings,
+                start,
+                end,
+            }
+        });
+
+        PosixTimeZone {
+            abbr,
+            offset,
+            transition_info,
+        }
+    }
+}
+
+impl PosixTimeZone {
+    pub fn to_string(&self) -> Result<String, core::fmt::Error> {
+        let mut posix_string = String::new();
+        write_abbr(&self.abbr, &mut posix_string)?;
+        write_inverted_time(&self.offset, &mut posix_string)?;
+
+        if let Some(transition_info) = &self.transition_info {
+            write_abbr(&transition_info.abbr, &mut posix_string)?;
+            if transition_info.savings != Time::one() {
+                write_inverted_time(&self.offset.add(transition_info.savings), &mut posix_string)?;
+            }
+            write_date_time(&transition_info.start, &mut posix_string)?;
+            write_date_time(&transition_info.end, &mut posix_string)?;
+        }
+        Ok(posix_string)
+    }
+}
+
+/// The representation of a POSIX time zone transition
+#[non_exhaustive]
+#[derive(Debug, PartialEq)]
+pub struct PosixTransition {
+    /// The transitions designated abbreviation
+    pub abbr: PosixAbbreviation,
+    /// The savings value to be added to the offset
+    pub savings: Time,
+    /// The start time for the transition
+    pub start: PosixDateTime,
+    /// The end time for the transition
+    pub end: PosixDateTime,
+}
+
+#[non_exhaustive]
+#[derive(Debug, PartialEq, Clone)]
+pub struct PosixAbbreviation {
+    /// Flag whether formatted abbreviation is numeric
+    pub is_numeric: bool,
```

Comment:
question: Sorry if this is a dump question... is numerical posix abbreviation defined somewhere else, I looked into the link, couldn't find the syntax (`<...>`)